src/commands/deploy/buildImageInAzure/UploadSourceCodeStep.ts

```diff
@@ -0,0 +1,41 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.md in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+import { getResourceGroupFromId } from '@microsoft/vscode-azext-azureutils';
+import { AzureWizardExecuteStep, nonNullValue } from '@microsoft/vscode-azext-utils';
+import * as fse from 'fs-extra';
+import * as tar from 'tar';
+import { createContainerRegistryManagementClient } from '../../../utils/azureClients';
+import { IBuildImageInAzureContext } from './IBuildImageInAzureContext';
+
+const vcsIgnoreList = ['.git', '.gitignore', '.bzr', 'bzrignore', '.hg', '.hgignore', '.svn'];
+
+export class UploadSourceCodeStep extends AzureWizardExecuteStep<IBuildImageInAzureContext> {
+    public priority: number = 175;
+
+    public async execute(context: IBuildImageInAzureContext): Promise<void> {
+        context.registryName = nonNullValue(context.registry?.name);
+        context.resourceGroupName = getResourceGroupFromId(nonNullValue(context.registry?.id));
+        context.client = await createContainerRegistryManagementClient(context);
+
+        const source: string = context.rootFolder.uri.fsPath;
+        let items = await fse.readdir(source);
+        items = items.filter(i => !(i in vcsIgnoreList));
+        tar.c({ cwd: source }, items).pipe(fse.createWriteStream(context.tarFilePath));
+
+        const sourceUploadLocation = await context.client.registries.getBuildSourceUploadUrl(context.resourceGroupName, context.registryName);
+        const uploadUrl: string = nonNullValue(sourceUploadLocation.uploadUrl);
+        const relativePath: string = nonNullValue(sourceUploadLocation.relativePath);
+
+        const storageBlob = await import('@azure/storage-blob');
```

Comment:
I think it might be better to have this outside of the class declaration. I think if we leave it in here, it keeps getting instantiated and we'll have more unnecessary async calls to import the module.

Reply:
Actually, if I remember correctly, when you await an import, it lazy-loads it and only does this once. So that means when it comes to this line of code again, the promised will already be resolved. So disregard my comment.